src/Microsoft.OData.Core/ODataRawInputContext.cs

```diff
@@ -115,7 +117,7 @@ internal override object ReadValue(IEdmPrimitiveTypeReference expectedPrimitiveT
         internal override Task<object> ReadValueAsync(IEdmPrimitiveTypeReference expectedPrimitiveTypeReference)
         {
             // Note that the reading is actually synchronous since we buffer the entire input when getting the stream from the message.
-            return TaskUtils.GetTaskForSynchronousOperation(() => this.ReadValueImplementation(expectedPrimitiveTypeReference));
+            return this.ReadValueImplementationAsync(expectedPrimitiveTypeReference);

```

Comment:
If this is actually synchronous, was it necessary to create an async implementation?

Reply:
Do you know at which point this buffering of the entire input takes place?